src/main/java/bio/terra/service/auth/iam/IamService.java

```diff
@@ -42,21 +41,22 @@
 public class IamService {
   private final Logger logger = LoggerFactory.getLogger(IamService.class);
 
-  private static final int AUTH_CACHE_SIZE_DEFAULT = 100;
+  private static final int AUTH_CACHE_TIMEOUT_SECONDS_DEFAULT = 60;
 
   private final IamProviderInterface iamProvider;
   private final ConfigurationService configurationService;
-  private final Map<AuthorizedCacheKey, AuthorizedCacheValue> authorizedMap;
+  private final Map<AuthorizedCacheKey, Boolean> authorizedMap;
 
   @Autowired
   public IamService(IamProviderInterface iamProvider, ConfigurationService configurationService) {
     this.iamProvider = iamProvider;
     this.configurationService = configurationService;
-    int cacheSize =
+    int ttl =
         Objects.requireNonNullElse(
-            configurationService.getParameterValue(AUTH_CACHE_SIZE), AUTH_CACHE_SIZE_DEFAULT);
+            configurationService.getParameterValue(AUTH_CACHE_TIMEOUT_SECONDS),
+            AUTH_CACHE_TIMEOUT_SECONDS_DEFAULT);
     // wrap the cache map with a synchronized map to safely share the cache across threads
-    authorizedMap = Collections.synchronizedMap(new LRUMap<>(cacheSize));
+    authorizedMap = Collections.synchronizedMap(new PassiveExpiringMap<>(ttl, TimeUnit.SECONDS));
```

Comment:
FWIW, we could also give the cache an upper bound on size by wrapping the `PassiveExpiringMap` around an `LRUMap`. I'm not sure if that's overkill or not.  It probably is.  At some point, we should add a thing to push metrics on cache size to prometheus (I created a ticket for it https://broadworkbench.atlassian.net/browse/DR-2650)

Reply:
Thank you for filing that metrics ticket!

We chatted about this over [Slack](https://broadinstitute.slack.com/archives/C01VBGH431S/p1655385668165999), but sharing here for visibility:

> My suspicion with the cache size setting was that it existed because we were using a LRUMap, and not that we were using a LRUMap to enforce the cache size.